test/integration/aaa_download_only_test.go

```diff
@@ -58,18 +60,32 @@ func TestDownloadOnly(t *testing.T) {
 					defer PostMortemLogs(t, profile)
 
 					// --force to avoid uid check
-					args := append([]string{"start", "--download-only", "-p", profile, "--force", "--alsologtostderr", fmt.Sprintf("--kubernetes-version=%s", v), fmt.Sprintf("--container-runtime=%s", r)}, StartArgs()...)
+					args := append([]string{"start", "-o=json", "--download-only", "-p", profile, "--force", "--alsologtostderr", fmt.Sprintf("--kubernetes-version=%s", v), fmt.Sprintf("--container-runtime=%s", r)}, StartArgs()...)
 
-					// Preserve the initial run-result for debugging
+					rt, err := Run(t, exec.CommandContext(ctx, Target(), args...))
 					if rrr == nil {
-						rrr, err = Run(t, exec.CommandContext(ctx, Target(), args...))
-					} else {
-						_, err = Run(t, exec.CommandContext(ctx, Target(), args...))
+						// Preserve the initial run-result for debugging
+						rrr = rt
 					}
-
 					if err != nil {
 						t.Errorf("failed to download only. args: %q %v", args, err)
 					}
+					s := bufio.NewScanner(bytes.NewReader(rt.Stdout.Bytes()))
```

Comment:
This could potentially be a Subtest
another t.Run with a name like "validateStepNumber"

so when it fails we know exaclty what part of download only test failed

example of adding subtests in TestStartStop test
https://github.com/medyagh/minikube/blob/b6ee8f2f6777921a87959df78e2497fafc9c5864/test/integration/start_stop_delete_test.go#L100

Reply:
Done